src/Build/BuildCheck/Infrastructure/BuildCheckCentralContext.cs

```diff
@@ -189,6 +196,7 @@ private void RunRegisteredActions<T>(
     where T : CheckData
     {
         string projectFullPath = checkData.ProjectFilePath;
+        List<CheckWrapper> checksToRemove = [];
```

Comment:
Let's allocate this only when we experience error.

We can be processing large amounts of data events (e.g. for property reads/writes - those might be very frequent) and in majority of builds there should be no exceptions during processing.